linux/deploy_vm/rebuild_unattend_install_iso.yml

```diff
@@ -28,19 +29,21 @@
   ansible.builtin.set_fact:
     unattend_install_file_name: "{{ new_unattend_install_conf | basename }}"
 
-- name: "Set fact of unattend install config file name"
-  ansible.builtin.set_fact:
-    unattend_install_os_type: "{{ unattend_install_conf.split('/')[0]  | lower }}"
+- name: "Rebuild ISO image file for {{ unattend_installer }}"
+  include_tasks: "rebuild_{{ unattend_installer.split('-')[0] | lower }}_unattend_install_iso.yml"
+  when: unattend_installer in ["Ubuntu-Ubiquity", "Ubuntu-Subiquity", "Photon", "Debian", "Pardus"]
 
-- name: "rebuild ISO for {{ unattend_install_os_type }}"
-  include_tasks: "rebuild_{{ unattend_install_os_type }}_unattend_install_iso.yml"
-  when: unattend_install_os_type in ["photon", "debian", "ubuntu", "pardus"]
-
-- name: "Rebuild ISO for FreeBSD"
+- name: "Rebuild ISO image file for FreeBSD"
   community.general.iso_customize:
     src_iso: "{{ src_iso_file_path }}"
     dest_iso: "{{ rebuilt_unattend_iso_path }}"
     add_files:
       - src_file: "{{ new_unattend_install_conf }}"
         dest_file: "/etc/installerconfig"
-  when: unattend_install_os_type == "freebsd"
+  when: unattend_installer == "FreeBSD"
+
+# Remove original OS ISO image downloaded from ESXi datastore
```

Comment:
Can this remove task also be put in `always` block?

Reply:
The reason is ?

Reply:
This temporary files cleanup can also be put in `always` part like others.

Reply:
This src ISO is only used for rebuild ISO image (for Debian, Pardus, Photon, Ubuntu Desktop 23.04 later, Ubuntu Server), it's an copy of original ISO not a new generated one. And this remove task is not a common task for deploy VM from ISOs (like RHEL, SLE, etc) and OVAs. So I would prefer to remove it just after using.

Reply:
`always` part in `deploy_vm.yml` is also used for all GOS install situations to cleanup cache folder, not for specific GOS or specific file, so no need to handle it here if decide to do this cleanup in `always`.
```
- name: "Clean up local cache"
   when:
        - unattend_iso_cache is defined
        - unattend_iso_cache
    block:
        - name: "Remove ISO files from local cache folder {{ unattend_iso_cache }}"
           include_tasks: ../../common/delete_local_file.yml
.... 
```